### DISCUSSION TITLE
add_includedirs添加路径后，无法识别代码中路径形式的头文件包含

### DISCUSSION THREAD

**Discussion by Jackey990709** [xmake-io/xmake#5593] [Q&A]
在lua文件中将所有参与编译的头文件的路径使用add_includedirs进行添加，代码文件中有很多类似 `#include "netif/zepif.h"`的头文件包含形式，使用xmake进行编译后会报文件不存在的错，但如果把它改成 `#include "zepif.h"`就不会报错，这两种编写方式在正常使用的IDE中皆可正常编译，使用xmake该如何正常编译这种含有路径形式的头文件包含的代码工程

**Comment by waruqi** [comment]
用 add_includedirs，然后路径对上

**Reply by Jackey990709** [reply]
比如说zepif.h的路径是"src/MCAL/netif/zepif.h",当工程里的写法是`#include "zepif.h"`时，xmake.lua里对应的是 `add_includedirs（"src/MCAL/netif"）`,此时xmake编译没有问题，当工程里的写法是`#include "netif/zepif.h"`时，我尝试在xmake.lua中加入 `add_includedirs（"src/MCAL"）`或者保持原样，皆会报No such file or directory

**Reply by star-hengxing** [reply]
检查 `xmake -rv` 里的 `-I` 开头的编译命令是否正确。

**Reply by Jackey990709** [reply]
编译命令是正确的，当工程里的写法是#include "zepif.h"时，是可以正常编译的

**Reply by star-hengxing** [reply]
当你用 `add_includedirs("src/MCAL")` 和 `"netif/zepif.h"` 的时候编译命令是怎么样的，然后自己手动命令行看看怎么改。。

**Reply by Jackey990709** [reply]
我尝试了，并没有解决，我的问题就是当头文件从`#include "zepif.h"`变为`#include "netif/zepif.h"`时，原本的xmake.lua文件该进行哪些修改才可正常编译

**Reply by star-hengxing** [reply]
这是很基础也是很简单的 api，各个构建系统之间没有啥差异的。建议你把项目发上来看看

**Reply by waruqi** [reply]
这种其实跟 xmake 也没啥关系，xmake 只保证你设置了 add_includedirs, 它正确帮你作为 `-Ixx` 传入编译，只要传入了，剩下的就是 编译器和和你项目代码如何 include 的事情。。


自己 `-v` 检查下编译命令中的 `-Ixx` 参数路径，以及你的 `#include ` 路径，自己调下就行了。。
